digitalocean/database/resource_database_opensearch_config.go

```diff
@@ -0,0 +1,533 @@
+			},
+			"thread_pool_write_size": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntAtLeast(1),
+			},
+			"thread_pool_write_queue_size": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntAtLeast(10),
+			},
+			"thread_pool_force_merge_size": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntAtLeast(1),
+			},
+			"override_main_response_version": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+			"script_max_compilations_rate": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"cluster_max_shards_per_node": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntAtLeast(100),
+			},
+			"cluster_routing_allocation_node_concurrent_recoveries": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntAtLeast(2),
+			},
+			"plugins_alerting_filter_by_backend_roles_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+			"reindex_remote_whitelist": {
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceDigitalOceanDatabaseOpensearchConfigCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+	clusterID := d.Get("cluster_id").(string)
+
+	if err := updateOpensearchConfig(ctx, d, client); err != nil {
+		return diag.Errorf("Error updating Opensearch configuration: %s", err)
+	}
+
+	d.SetId(makeDatabaseOpensearchConfigID(clusterID))
+
+	return resourceDigitalOceanDatabaseOpensearchConfigRead(ctx, d, meta)
+}
+
+func resourceDigitalOceanDatabaseOpensearchConfigUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+
+	if err := updateOpensearchConfig(ctx, d, client); err != nil {
+		return diag.Errorf("Error updating Opensearch configuration: %s", err)
+	}
+
+	return resourceDigitalOceanDatabaseOpensearchConfigRead(ctx, d, meta)
+}
+
+func updateOpensearchConfig(ctx context.Context, d *schema.ResourceData, client *godo.Client) error {
+	clusterID := d.Get("cluster_id").(string)
+
+	opts := &godo.OpensearchConfig{}
+
+	if v, ok := d.GetOk("ism_enabled"); ok {
+		opts.IsmEnabled = godo.PtrTo(v.(bool))
+	}
+
+	if v, ok := d.GetOk("ism_history_enabled"); ok {
+		opts.IsmHistoryEnabled = godo.PtrTo(v.(bool))
+	}
+
+	if v, ok := d.GetOk("ism_history_max_age_hours"); ok {
+		opts.IsmHistoryMaxAgeHours = godo.PtrTo(v.(int))
+	}
+
+	if v, ok := d.GetOk("ism_history_max_docs"); ok {
+		opts.IsmHistoryMaxDocs = godo.PtrTo(v.(int64))
```

Comment:
```suggestion
		opts.IsmHistoryMaxDocs = godo.PtrTo(int64(v.(int)))
```
I'm seeing a panic here:

```
Stack trace from the terraform-provider-digitalocean plugin:

panic: interface conversion: interface {} is int, not int64

goroutine 31 [running]:
github.com/digitalocean/terraform-provider-digitalocean/digitalocean/database.updateOpensearchConfig({0x12911d0, 0xc000159500}, 0xc000177e00, 0xc0003b4f08)
        /home/asb/projects/go/src/github.com/digitalocean/terraform-provider-digitalocean/digitalocean/database/resource_database_opensearch_config.go:294 +0x1aa5
```